src/adonis5-ally-okta/index.ts

```diff
@@ -142,23 +145,82 @@ export class YourDriver extends Oauth2Driver<YourDriverAccessToken, YourDriverSc
    * is made by the base implementation of "Oauth2" driver and this is a
    * hook to pre-configure the request.
    */
-  // protected configureRedirectRequest(request: RedirectRequest<YourDriverScopes>) {}
+
+  protected configureRedirectRequest(request: RedirectRequest<OktaDriverScopes>) {
+    request.param('scope', this.config.scopes)
+    request.param('state', new Date().getTime())
```

Comment:
Hi @r-herrera, does _state_ and _nonce_ supposed to have the same value? If so, don't you think it's better if we capture the value returned by _getTime()_ rather than calling it for each assignment?

Reply:
Other than that. LGTM!

Reply:
Hi @timothyryeyn-cupa   thanks for pointing it out. I realized that nonce and state serve different purposes and shouldn't have the same value. I'll adjust the implementation accordingly.